woocommerce/woocommerce
# Fix Tax Analytics total results & export

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



Add group_by clause for total_results calculation

Fixes
- https://github.com/woocommerce/woocommerce/issues/48841
- https://github.com/woocommerce/woocommerce/issues/49635

The total_results calculation used SQL statement w/o GROUP BY clause used for the final list.
This resulted in the total items number mismatch, giving additional items to the table and blocking export as the percentage was >100.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Use the `trunk` first, to make sure faulty scenario is in place
2. Place an order with a tax, let's say `PL 22% TAX-PL 1`
3. Go to `/wp-admin/admin.php?page=wc-settings&tab=tax&section=standard`
4. Change the value of that tax `PL 18% TAX-PL 1`, 
5. Place another order with updated tax
6. Go to `/wp-admin/admin.php?page=wc-admin&path=%2Fanalytics%2Ftaxes`
7. Select a period that covers the orders
8. Go to the last page of the table
![image](https://github.com/user-attachments/assets/374b4667-668b-4e0e-8eb0-bac1d87dee3b)
9. Check the network tab in dev tools. Check `taxes` request. `x-wp-total:` header do not match the number of results in the response
10. Click "Download"
11. Wait for the export action to be scheduled
12. Check that no email was sent
13. Checkout this branch
14. Redo steps 6-11 above
15. checking that the numbers in header and response are correct
16. Check that clicking on Download sends the export via email correctly.





### Additional notes

There is still a mismatch in rendered number of `N tax codes` which is the result of mismatch between `taxes` and `.../stats` endpoints. https://github.com/woocommerce/woocommerce/pull/51218

### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
